src/main/java/com/epam/webapphello/connection/ConnectionPool.java

```diff
@@ -0,0 +1,86 @@
+package com.epam.webapphello.connection;
+
+import com.epam.webapphello.exception.ConnectionException;
+
+import java.sql.SQLException;
+import java.util.ArrayDeque;
+import java.util.Queue;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class ConnectionPool {
+
+    private static volatile ConnectionPool instance;
+    private Queue<ProxyConnection> availableConnection;
+    private Queue<ProxyConnection> connectionInUse;
+    private static final int CONNECTIONS_NUMBER = 10;
+    private static final Lock LOCK_INSTANCE = new ReentrantLock();
+    private ConnectionFactory connectionFactory = new ConnectionFactory();
+
+
+    private ConnectionPool() {
+    }
+
+
+    public static ConnectionPool getInstance() {
+        ConnectionPool localInstance = instance;
+        if (localInstance == null) {
+            LOCK_INSTANCE.lock();
+            localInstance = instance;
+            if (localInstance == null) {
+                instance = localInstance = new ConnectionPool();
+            }
+            LOCK_INSTANCE.unlock();
+        }
+        return localInstance;
+    }
+
+
+    public void returnConnection(ProxyConnection proxyConnection) {
+        LOCK_INSTANCE.lock();
```

Comment:
use different lock, not isntance lock

Reply:
you mean to change name or create a new non static lock for this non static methods

Reply:
?

Reply:
static or no